letsencrypt/tasks/main.yml

```diff
@@ -225,3 +225,17 @@
       {{ letsencrypt_command }} renew --agree-tos --post-hook='service nginx reload'
   tags:
     - letsencrypt
+
+- name: Create a cleanup cronjob to clear old key and csr files
+  # see the description in defaults/main.yml
+  # only very rare situations (like Tahoe) will need this.
+  cron:
+    name: "letsencrypt disk cleanup"
+    cron_file: certbot_appsembler
+    special_time: daily
+    user: root
+    job: >
+      find /etc/letsencrypt/{keys,csr,archive} -type f -mtime +{{ letsencrypt_cleanup_days }} -exec rm {} \;
```

Comment:
I don't trust the filesystem for such critical information. The certificate itself have expiry information.

My question is:

 - Do you trust the filesystem esp. GCP Fuse to have correct datatime values?
 - Is it worth going to the extra mile and writing a proper script to check the certificate info?
 - Any check to ensure that the cert is not being used at the moment?

Reply:
@OmarIthawi I started down that path. Wrote this script to explore the `/etc/letsencrypt/archive` directory: https://gist.github.com/thraxil/035b8bb53c6649c1407d6a7e9c0eb2e6

Ultimately though, it would delete at most 568 files from the archive dir (if we deleted every cert file that isn't currently in use, not just ones more than 90 or 180 days old).

Meanwhile, `keys`, and `csr` directories have 3800 files in them that in theory are *all* completely safe to delete. So it just seems like low hanging fruit to prune the old ones there and just not invest too much time in the `archive` directory.

Reply:
:heart: